embed-code-go/fragmentation/fragmentation.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	absoluteCodeFile, err := filepath.Abs(codeFileRelative)
+	fragmentation.CodeFile = absoluteCodeFile
+	if err != nil {
+		panic(err)
+	}
+
+	fragmentation.Configuration = config
+
+	return fragmentation
+}
+
+//
+// Public methods
+//
+
+// Splits the file into fragments.
+//
+// Returns a refined content of the file to be cut into fragments, and the Fragments.
+func (fragmentation Fragmentation) Fragmentize() ([]string, map[string]Fragment, error) {
+	fragmentBuilders := make(map[string]*FragmentBuilder)
+	var contentToRender []string
+
+	file, err := os.Open(fragmentation.CodeFile)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text() + "\n"
+		contentToRender, fragmentBuilders, err = fragmentation.parseLine(line, contentToRender, fragmentBuilders)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	fragments := make(map[string]Fragment)
+	for k, v := range fragmentBuilders {
+		fragments[k] = v.Build()
+	}
+	fragments[DefaultFragment] = CreateDefaultFragment()
+
+	return contentToRender, fragments, nil
+
+}
+
+// Serializes fragments to the output directory.
+//
+// Keeps the original directory structure relative to the sources root dir.
+// That is, `SRC/src/main` becomes `OUT/src/main`.
+func (fragmentation Fragmentation) WriteFragments() error {
+	allLines, fragments, err := fragmentation.Fragmentize()
+	if err != nil {
+		return err
+	}
+
+	EnsureDirExists(fragmentation.targetDirectory())
+
+	for _, fragment := range fragments {
+		fragmentFile := NewFragmentFileFromAbsolute(fragmentation.CodeFile, fragment.Name, fragmentation.Configuration)
+		fragment.WriteTo(fragmentFile, allLines, fragmentation.Configuration)
+	}
+	return nil
+}
+
+//
+// Static functions
+//
+
+// Searches for code files with patterns defined in configuration
+// and fragmentizes them with creating fragmented files as a result.
+func WriteFragmentFiles(configuration configuration.Configuration) error {
+	includes := configuration.CodeIncludes
+	codeRoot := configuration.CodeRoot
+	for _, rule := range includes {
+		pattern := fmt.Sprintf("%s/%s", codeRoot, rule)
+		codeFiles, _ := filepath.Glob(pattern)
+		for _, codeFile := range codeFiles {
+			if ShouldFragmentize(codeFile) {
+				fragmentation := NewFragmentation(codeFile, configuration)
+				err := fragmentation.WriteFragments()
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// Reports whether the file stored at the filePath exists, is file and is textual-encoded.
+func ShouldFragmentize(fileName string) bool {
+	info, err := os.Stat(fileName)
+	if err != nil {
+		panic(err)
+	}
+
+	isFile := !info.IsDir()
+	isValidEncoding := IsEncodedAsText(fileName)
+
+	return isFile && isValidEncoding
+}
+
+//
+// Private methods
+//
+
+// This function parses a single line of input and performs the following actions:
+//   - Identifies fragment start and end markers within the line;
```

Comment:
Given the layout of this list (after `:`, each statement ends with `;`), points should start with a small letter.